search-index/text2embeddings.py

```diff
@@ -18,10 +19,108 @@
 logger = get_logger(__name__)
 
 
+def get_text_from_list(text_block: dict, prev_processed_text_block: dict) -> str:
```

Comment:
Tried a bunch of stuff here, but eventually settled on this ridiculously simple function as more complex implementations just created more headaches than they solved. See PR comments for more context.